<?xml version="1.0" encoding="UTF-8" ?>

<launch>
    <env name="ROSCONSOLE_FORMAT" value="[${severity}][${time:%y-%m-%d %H:%M}]: ${message}"/>
    <!-- <include file="$(find floor_plane_hough)/launch/fpr.launch"/> -->
    <include file="$(find floor_plane_regression)/launch/fpr.launch"/>
    
    <node name="task_server" pkg="floor_nav" type="task_server" output="screen" >
        <param name="lib_path" value="$(find floor_nav)/../../devel/share/floor_nav/tasks"/>
        <param name="base_frame" value="bubbleRob"/>
        <param name="reference_frame" value="world"/>
        <!--<param name="base_frame" value="body"/>
        <param name="reference_frame" value="map"/>-->
    </node>
    
    <node pkg="floor_plane_mapping" type="floor_plane_mapping" name="floor_plane_mapping" output="screen">
        launch-prefix="xterm -e gdb --args"
        <param name="n_x" value="10" />
        <param name="x_min" value="-4.5" />
        <param name="x_max" value="4.5" />
        <param name="n_y" value="10" />
        <param name="y_min" value="-4.5" />
        <param name="y_max" value="4.5" />
        <param name="angle_threshold" value="8.0" />
        <!-- <remap from="/floor_plane_mapping/floor_param" to="/floor_plane_hough/floor_plane_param"/> -->
        <remap from="/floor_plane_mapping/floor_param" to="/floor_plane_regression/floor_plane_param"/>
  </node>
  
    <node name="viz_graph" pkg="floor_graph" type="viz_graph.py" output="screen"> </node>
    <node name="image_view" pkg="image_view" type="image_view" respawn="false" output="screen">
        <remap from="image" to="/floor_plane_mapping/map_img" />
    </node>
    <node pkg="floor_plane_mapping" name="mission1" type="mission1.py" output="screen"> </node>

</launch>

